File: 1 Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
            var hs = new Dictionary<int, int>();
            for (int i = 0; i < nums.Length; i++)
                if (hs.ContainsKey(nums[i]))
                {
                    if (nums[i] * 2 == target) return new int[] { hs[nums[i]], i };
                }
                else hs.Add(nums[i], i);
            for (int i = 0; i < nums.Length; i++) if (hs.ContainsKey(target - nums[i]) && hs[target - nums[i]] != i)
                    return new int[] { i, hs[target - nums[i]] };
            return new int[] { };
    }
}

File: 1 Two Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pos = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (pos.containsKey(target - num)) {
                return new int[]{pos.get(target - num), i};
            }
            pos.put(num, i);
        }
        return null;
    }
}
